e2e/callback_test.go

```diff
@@ -195,351 +196,322 @@ func (suite *CbTestSuite) SetupTest() {
 
 	confirmNftContracts(suite.chainA, suite.chainC, suite.testerA.String(), suite.testerC.String(), suite.cw721A.String(), ACCw721)
 
 	// Let send back all NFTs to Chain A
+	// C -> A token_id 4
 	sendIcsFromChainCToA(suite, ACCw721, "4", memo, true)
 	confirmNftContracts(suite.chainC, suite.chainA, suite.testerC.String(), suite.testerA.String(), ACCw721, suite.cw721A.String())
 
+	// C -> B token_id 2
 	sendIcsFromChainCToB(suite, BCCw721, "2", memo, true)
 	confirmNftContracts(suite.chainC, suite.chainB, suite.testerC.String(), suite.testerB.String(), BCCw721, suite.cw721B.String())
-
+	// B -> A again token_id 2
 	sendIcsFromChainBToA(suite, suite.cw721B.String(), "2", memo, true)
 	confirmNftContracts(suite.chainB, suite.chainA, suite.testerB.String(), suite.testerA.String(), suite.cw721B.String(), suite.cw721A.String())
 
-	chainAOwner1 := queryGetOwnerOf(suite.T(), suite.chainA, suite.cw721A.String(), "2")
+	chainAOwner1 := test_suite.QueryGetOwnerOf(suite.T(), suite.chainA, suite.cw721A.String(), "2")
 	require.Equal(suite.T(), chainAOwner1, suite.testerA.String())
-	chainAOwner2 := queryGetOwnerOf(suite.T(), suite.chainA, suite.cw721A.String(), "4")
+	chainAOwner2 := test_suite.QueryGetOwnerOf(suite.T(), suite.chainA, suite.cw721A.String(), "4")
 	require.Equal(suite.T(), chainAOwner2, suite.testerA.String())
 
 	// NFTs should no exist on Chain B and Chain C, they should be burned and query for owner should error
-	err := queryGetOwnerOfErr(suite.T(), suite.chainB, suite.cw721B.String(), "2")
+	err := test_suite.QueryGetOwnerOfErr(suite.T(), suite.chainB, suite.cw721B.String(), "2")
 	require.Error(suite.T(), err)
-	err = queryGetOwnerOfErr(suite.T(), suite.chainC, BCCw721, "2")
+	err = test_suite.QueryGetOwnerOfErr(suite.T(), suite.chainC, BCCw721, "2")
 	require.Error(suite.T(), err)
-	err = queryGetOwnerOfErr(suite.T(), suite.chainC, ACCw721, "4")
+	err = test_suite.QueryGetOwnerOfErr(suite.T(), suite.chainC, ACCw721, "4")
 	require.Error(suite.T(), err)
 }
+
+func callbackMemo(srcCallback, srcReceiver, dstCallback, dstReceiver string) string {
+	srcCallback = test_suite.ParseOptional(srcCallback)
+	srcReceiver = test_suite.ParseOptional(srcReceiver)
+	dstCallback = test_suite.ParseOptional(dstCallback)
+	dstReceiver = test_suite.ParseOptional(dstReceiver)
+	memo := fmt.Sprintf(`{ "callbacks": { "ack_callback_data": %s, "ack_callback_addr": %s, "receive_callback_data": %s, "receive_callback_addr": %s } }`, srcCallback, srcReceiver, dstCallback, dstReceiver)
+	return b64.StdEncoding.EncodeToString([]byte(memo))
+}
+
+func nftSentCb() string {
+	return b64.StdEncoding.EncodeToString([]byte(`{ "nft_sent": {}}`))
+}
+
+func nftReceivedCb() string {
+	return b64.StdEncoding.EncodeToString([]byte(`{ "nft_received": {}}`))
+}
+
+func failedCb() string {
+	return b64.StdEncoding.EncodeToString([]byte(`{ "fail_callback": {}}`))
+}
+
+func sendIcsFromChainAToB(suite *CallbackTestSuite, nft, tokenId, memo string, relay bool) {
```

Comment:
some naming suggestions:
- `sendIcsFromChainAToB` -> `sendNftFromChainAToB`
- `nft` -> `nftContract` (or `nft_contract`, since  I dunno what's the naming convention in Go is